quinn-proto/src/connection.rs

```diff
@@ -31,6 +31,10 @@ impl From<ConnectionHandle> for usize {
 }
 
 pub struct Connection {
+    // TODO: Is there a better name for this?  I feel like there is, but I don't know enough about
+    // the way this struct is used to know if the knowledge of "app" makes sense here.
+    // Just `closed`?
+    pub app_closed: bool,
```

Comment:
There's a bunch of different states a connection can be in that qualify as "closed" (see `State::is_closed`). This state is for tracking whether the next layer up has explicitly called `close`, which is used as a signal that it will not make any further calls with that `ConnectionHandle`, and so its resources may be freed as soon as internal logic and the peer no longer needs it.